File: sources/1759.cpp
#include <bits/stdc++.h>
using namespace std;

int L, C;
vector<char> v;
int acnt, bcnt;
char ret[17];
int visited[17];

void dfs(int cx, int cnt) {
    int k = 0;
    visited[cx] = 1;
    if (cnt == L - 1) {
        for (int i = 0; i < C; i++) {
            if (visited[i] == 1) {
                ret[k] = v[i];
                k++;
            }
        }
        for (int i = 0; i < L; i++) {
            if (ret[i] == 'a' || ret[i] == 'e' || ret[i] == 'i' || ret[i] == 'o' || ret[i] == 'u') acnt++;
            else bcnt++;
        }
        if (acnt >= 1 && bcnt >= 2) {
            for (int i = 0; i < L; i++) printf("%c", ret[i]);
            printf("\n");
        }
        acnt = 0;
        bcnt = 0;
        k = 0;
        return ;
    }
    for (int i = cx+1; i < C; i++) {
        visited[i] = 1;
        dfs(i, cnt+1);
        visited[i] = 0;
    }
    visited[cx] = 0;
}

int main() {
    char alp;
    scanf("%d %d", &L, &C);
    for (int i = 0; i < C; i++) {
        getchar();
        scanf("%c", &alp);
        v.push_back(alp);
    }
    sort(v.begin(), v.end());
    for (int i = 0; i <= C-L+1; i++) {
        dfs(i, 0);
    }
    
    return 0;
}

File: sources/python/1759.py
import sys
input = sys.stdin.readline

def isVow(a):
	if a == 'a' or a == 'e' or a == 'i' or a == 'o' or a == 'u':
		return True
	else:
		return False

L, C = map(int, input().split())
possible = input().split()
visited = [False] * C
possible.sort()
def dfs(cx, cnt):
	if cnt == L - 1:
		TC = []
		c = 0
		v = 0
		for i in range(C):
			if visited[i]:
				TC.append(possible[i])
		for i in range(len(TC)):
			if isVow(TC[i]):
				v += 1
			else:
				c += 1
		if c >= 2 and v >= 1:
			for i in range(len(TC)):
				print(TC[i], end='')
			print('')
		return 0

	visited[cx] = True
	for i in range(cx+1, C):
		visited[i] = True
		dfs(i, cnt+1)
		visited[i] = False
	visited[cx] = False

for i in range(0, C):
	dfs(i, 0)


	
	
